h/api/test/views_test.py

```diff
@@ -99,6 +101,34 @@ def test_create(mock_create_annotation, user):
     _assert_event_published('create')
 
 
+@patch("h.api.views.Annotation.save")
+@patch("h.api.views.nipsa.has_nipsa")
+@patch("h.api.views.get_user")
+def test_create_adds_nipsa_flag(get_user, has_nipsa, _):
+    get_user.return_value = Mock(
+        id="test_id", consumer=Mock(key="test_key"))
+    has_nipsa.return_value = True
+    request = DummyRequest(json_body=_new_annotation)
+
+    annotation = views.create(request)
+
+    assert annotation.get("nipsa") is True
+
+
+@patch("h.api.views.Annotation.save")
+@patch("h.api.views.nipsa.has_nipsa")
+@patch("h.api.views.get_user")
+def test_create_does_not_add_nipsa_flag(get_user, has_nipsa, _):
+    get_user.return_value = Mock(
+        id="test_id", consumer=Mock(key="test_key"))
+    has_nipsa.return_value = False
+    request = DummyRequest(json_body=_new_annotation)
+
+    annotation = views.create(request)
+
+    assert not annotation.get("nipsa")
```

Comment:
Hmm, I would have thought Es would play fine with that (the Es filter is a positive one: filter out anything that does have `nipsa: true`). But I guess what I really want here (if we want to test specifically what the actual behaviour is) is `"nipsa" not in annotation`